GrpcClient/EmployeeGrpcRepository.cs

```diff
@@ -0,0 +1,286 @@
+                Password = employee.Password
+            });
+            //converting
+            List<ShiftDTO> grpcShiftDtos = reply.AssignedShifts.Dtos.ToList();
+            List<Entities.Shift> shifts = GrpcShiftDtosToEntityShiftsList(grpcShiftDtos);
+            Entities.Employee employeeRecieved = grpcEmployeeDtoToEntityEmployee(reply, shifts);
+        
+            return employeeRecieved;
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.AlreadyExists)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+
+
+
+        return null;
+    }
+
+    public async Task<Entities.Employee> UpdateAsync(Entities.Employee employee)
+    {
+        Entities.Employee employee2 = new Entities.Employee();
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.UpdateSingleEmployeeAsync(new UpdateEmployeeDTO
+            {
+                Id = employee.Id,
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                WorkingNumber = UInt32.CreateChecked(employee.WorkingNumber),
+                Email = employee.Email,
+                Password = employee.Password
+            });
+            employee2 = grpcEmployeeDtoToEntityEmployee(reply,
+                GrpcShiftDtosToEntityShiftsList(reply.AssignedShifts.Dtos.ToList()));
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+            else if (e.StatusCode == StatusCode.AlreadyExists)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+        return employee2;
+    }
+
+    public async Task DeleteAsync(long id)
+    {
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.DeleteSingleEmployeeAsync(new Id { Id_ = id });
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+    }
+
+    public IQueryable<Entities.Employee> GetManyAsync()
+    {
+        List<Entities.Employee> employees = new List<Entities.Employee>();
+        
+        using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+        var client = new Employee.EmployeeClient(channel);
+        var reply = client.GetAllEmployees(new Empty());
+        employees = grpcEmployeeDtoListToEntityEmployeeList(reply);
+       
+        return employees.AsQueryable();
+    }
+
+    public async Task<Entities.Employee> GetSingleAsync(long id)
```

Comment:
Comment prob to ignore, but we have a lot of super descriptive method names so shouldn't this one be GetSingleEmployeeAsync or GetSingleEmployeeByIdAsync to not confuse it?

Reply:
We are good with using just getsingle

Reply:
i somewhat agree with Karolina's idea but lets not get too silly with the names lol

getsingle has no other real way of happening without having an id